### ISSUE TITLE
[BUG] fatal error C1001: Internal compiler error.

### ISSUE THREAD

**Issue by edunad** [stephenberry/glaze#1123]
Not sure what is happening, but since the new versions i'm now getting

```
[build] D:\<folder name>\.cache\cpm\glaze\27bbdd96fd3142512b169ac3a4ce5040c94972cb\include\glaze/json/write.hpp(1308): fatal error C1001: Internal compiler error.
[build] (compiler file 'D:\a\_work\1\s\src\vctools\Compiler\CxxFE\sl\p1\c\symbols.c', line 32907)
[build]  To work around this problem, try simplifying or changing the program near the locations listed above.
[build] If possible please provide a repro here: https://developercommunity.visualstudio.com 
[build] Please choose the Technical Support command on the Visual C++ 
```

When trying to use `glz::write` with a struct

```c++
struct WorkshopModConfig {
		uint32_t type;

		std::string title;
		std::string version;
		std::string author;
	};
```

```c++
std::string buffer;
auto ec = glz::write<glz::opts{.comments = true, .error_on_unknown_keys = true, .skip_null_members = true, .prettify = false}>(settings, buffer);
if (ec) throw _logger->error("Failed to serialize mod settings");
```

If i comment out these lines, the compiler works fine.
I'm using `c++23` and `Vistual Studio Build Tools 2022`

**Comment by ConnerRose** [comment]
For me, since I defined a constructor for the struct I was writing from I needed to add explicit metadeta to handle the serialization. See https://github.com/stephenberry/glaze?tab=readme-ov-file#explicit-metadata.

**Comment by stephenberry** [comment]
Thanks for reporting this. I'll aim to fix this soon. Typically these MSVC internal compiler errors aren't too hard to fix. I just keep adding unit tests to catch more of them.

**Comment by stephenberry** [comment]
This issue turns out to be a missing guard for reflected structs when `.comments = true`, because reflection doesn't support built in comments. Will be merging a fix soon (#1124). If you set `.comments = false` your code should also run.
